server/api/reviews.js

```diff
@@ -0,0 +1,85 @@
+const express = require('express')
+const router = express.Router()
+const { Review, User, Order, OrderItem } = require('../db')
+
+// GET /api/reviews/:artworkId
+router.get('/:artworkId', async (req, res, next) => {
+  try {
+    const reviews = await Review.findAll({
+      where : {
+        artworkId: req.params.artworkId
+      }
+    })
+    res.send(reviews)
+  }
+  catch(err) {
+    console.log(err)
+    next(err);
+  }
+})
+
+// GET /api/reviews/user/:userId
+router.get('/user/:userId', async (req, res, next) => {
+  try {
+    const user = await User.findByPk(req.params.userId, {
+      include: [Review]
+    })
+    res.send(user.reviews)
+  }
+  catch(err) {
+    console.log(err)
+    next(err);
+  }
+})
+
+router.post('/', async (req, res, next) => {
+  try {
+    //make sure user is logged in
+    if (req.user) {
+      const userId = req.user.id
+      const { artwork, rating, text } = req.body
+      //find the user and include their orders and orderItems
+      const user = await User.findByPk(userId, {
+        include: {
+          model: Order,
+          include: [OrderItem]
+        }
+      })
+      //list of artIds they've ordered
+      const userOrderedArtIds = user.orders.reduce((acc,order)=>{
+        const artworks = order.orderItems.map(oi=>oi.artworkId)
+        acc.push(artworks)
+        return acc
+      },[]).flat()
+      //if they've ordered the art they're reviewing then...
+      if(userOrderedArtIds.includes(artwork.id)) {
+        const [newReview, created] = await Review.findOrCreate({
```

Comment:
i'm confused by the use of findOrCreate here... there doesnt seem to be a way to edit reviews in the UI unless i'm completely missing it, so shouldnt we only need to create them?

Reply:
I think the next few lines of code make it clearer. Don't want a user to leave multiple reviews for same art.